### PR TITLE
BUG: plotting grouped_hist with a single row frame #10214

### LINKED ISSUE #10214: hist raises AttributeError: 'AxesSubplot' object has no attribute 'ndim' with a one row DataFrame
Hello,

this code works fine:

    import pandas as pd
    import numpy as np
    import matplotlib.pyplot as plt
    bins = np.arange(80, 100 + 2, 1)
    df = pd.DataFrame({"Name": ["AAA", "BBB"], "ByCol": [1, 2], "Mark": [85, 89]})
    df

       ByCol  Mark Name
    0      1    85  AAA
    1      2    89  BBB

    df["Mark"].hist(by=df["ByCol"], bins=bins)
    plt.show()

but same code with a one row DataFrame such as

    df = pd.DataFrame({"Name": ["AAA"], "ByCol": [1], "Mark": [85]})
    df

       ByCol  Mark Name
    0      1    85  AAA

raises:

    ---------------------------------------------------------------------------
    AttributeError                            Traceback (most recent call last)
    <ipython-input-35-93bcd47d24f2> in <module>()
    ----> 1 df["Mark"].hist(by=df["Mark"], bins=bins)

    //anaconda/lib/python2.7/site-packages/pandas/tools/plotting.pyc in hist_series(self, by, ax, grid, xlabelsize, xrot, ylabelsize, yrot, figsize, bins, **kwds)
       2847                             **kwds)
       2848
    -> 2849     if axes.ndim == 1 and len(axes) == 1:
       2850         return axes[0]
       2851     return axes

    AttributeError: 'AxesSubplot' object has no attribute 'ndim'

Kind regards

### PR DESCRIPTION
**Pull request by s-celles**
closes #10214

### CODE REVIEW

**Comment by sinhrks** [comment]
Thanks! I think it can be in 0.16.2 (being released before 0.17.0), if you move release note description from "0.17.0.txt" to "0.16.2.txt" locates on the same directory. If you have any trouble, please feel free to ask.

**Comment by scls19fr** [comment]
I did it https://github.com/scls19fr/pandas/tree/hist-ndim
How should I submit my changes in same PR ?

**Comment by scls19fr** [comment]
I should fix https://github.com/pydata/pandas/issues/10214

File: pandas/tests/test_graphics.py [hunk 1]
```diff
@@ -3896,6 +3896,13 @@ def test_plotting_with_float_index_works(self):
         df.groupby('def')['val'].apply(lambda x: x.plot())
         tm.close()
 
+    def test_hist_single_row(self):
+        bins = np.arange(80, 100 + 2, 1)
```
Comment (jreback) [inline pandas/tests/test_graphics.py]: add the issue number as a comment here

File: pandas/tools/plotting.py [hunk 2]
```diff
@@ -2846,8 +2846,9 @@ def hist_series(self, by=None, ax=None, grid=True, xlabelsize=None,
                             xlabelsize=xlabelsize, xrot=xrot, ylabelsize=ylabelsize, yrot=yrot,
                             **kwds)
 
-    if axes.ndim == 1 and len(axes) == 1:
-        return axes[0]
+    if isinstance(axes, np.ndarray):
```
Comment (jreback) [inline pandas/tools/plotting.py]: this should just be ``hasattr(axes, 'ndim')``

**Comment by jreback** [comment]
pls see contribution guidelines [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

to update, you simply push a new commit. don't open and close different PR's that is confusing.

if you squash/rebase, then ``git push user-name branch-name -f``

**Comment by jreback** [comment]
pls squash.  @TomAugspurger ok with you?

**Comment by TomAugspurger** [comment]
Just gave a quick look. I think it's ok.

**Comment by scls19fr** [comment]
I'm quite busy. Could you squash ? Thanks

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/a8b5798b31f04705bd421a2902c523f927eb6b7e

thanks!

**Pull request closed without merging**